### ISSUE TITLE
Bug in add_param - functionality for tensorboard (scatter matrix view)

### ISSUE THREAD

**Issue by zero-lummel** [pytorch/pytorch#32528]
## 🐛 Bug

The add_param() functionality in torch.utils.tensorboard doesn't keep the type of the hyperparameter when writing them to tensorboard file. As a result the hparams are ordered in a alphabetic order instead of a numeric one (if the hparm type is float for instance). Another resulting behavior is that tensorboard is producing a single tick for each hparam on x-axis. 

## To Reproduce

1. Just use the example which is provided in the docu but change the range to 15 i.e.:
from torch.utils.tensorboard import SummaryWriter
with SummaryWriter() as w:
    for i in range(15):
        w.add_hparams({'lr': 0.1*i, 'bsize': i},
                      {'hparam/accuracy': 10*i, 'hparam/loss': 10*i})

2. Start tensorboard and got to hparam section. Then go to scatter plot matrix view and take a look on the x axis of bsize. 

## Expected behavior

Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 10 Pro
GCC version: (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0
CMake version: Could not collect

Python version: 3.6
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] numpydoc==0.9.2
[pip] torch==1.4.0
[pip] torchvision==0.5.0
[conda] blas                      1.0                         mkl
[conda] mkl                       2019.4                      245
[conda] mkl-service               2.3.0            py36hb782905_0
[conda] mkl_fft                   1.0.15           py36h14836fe_0
[conda] mkl_random                1.1.0            py36h675688f_0
[conda] pytorch                   1.4.0           py3.6_cuda101_cudnn7_0    pytorch
[conda] torch                     1.0.1                    pypi_0    pypi
[conda] torchvision               0.2.2.post3              pypi_0    pypi
## Additional context

![image](https://user-images.githubusercontent.com/60213215/72966813-15624b80-3dc0-11ea-81a1-0d8c6712921b.png)

**Comment by lanpa** [comment]
This should be solved in https://github.com/pytorch/pytorch/pull/31544

**Comment by taras-janea** [comment]
@zero-lummel, could you please confirm whether the issue is still reproducible on your setup with the latest PyTorch version?

If the problem persists, please share:
- The output of [this script](https://raw.githubusercontent.com/pytorch/pytorch/main/torch/utils/collect_env.py);
- Steps to reproduce the issue (ideally, a reproducer script).

**Comment by taras-janea** [comment]
Closing this issue due to inactivity. If you continue to experience this problem or have any related concerns, please don't hesitate to open a new issue.
